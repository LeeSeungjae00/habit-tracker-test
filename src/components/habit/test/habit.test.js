import React from 'react';
import { render, screen } from "@testing-library/react";
import userEvent from "@testing-library/user-event";
import '@testing-library/jest-dom'
import renderer from 'react-test-renderer'
import Habit from '../habit';

describe('habit', () => {
    const habit = { count: 0, id: 10, name: 'testing' }
    let onIncrement;
    let onDecrement;
    let onDelete;
    let HabitComponent;
    beforeEach(() => {
        onIncrement = jest.fn();
        onDecrement = jest.fn();
        onDelete = jest.fn();
        HabitComponent = (<Habit
            habit={habit}
            onIncrement={onIncrement}
            onDecrement={onDecrement}
            onDelete={onDelete}
        />)

    })
    it('renders', () => {
        const component = renderer.create(HabitComponent);
        expect(component.toJSON).toMatchSnapshot();
    })
    describe('button click', () => {
        beforeEach(() => {
            render(HabitComponent);
        })
        it('call on Increment when click + button', () => {
            const button = screen.getByTitle('increase');
            userEvent.click(button);
            expect(onIncrement).toHaveBeenCalledWith(habit)
        })
        it('call on Increment when click - button', () => {
            const button = screen.getByTitle('decrese');
            userEvent.click(button);
            expect(onDecrement).toHaveBeenCalledWith(habit)
        })
        it('call on Increment when click trash button', () => {
            const button = screen.getByTitle('delete');
            userEvent.click(button);
            expect(onDelete).toHaveBeenCalledWith(habit)
        })
    })

})