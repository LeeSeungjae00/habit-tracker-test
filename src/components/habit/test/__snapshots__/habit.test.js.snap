// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`habit renders 1`] = `[Function]`;
